game/squadrons/airwing.py

```diff
@@ -10,18 +10,38 @@
 from .squadron import Squadron
 from ..theater import ControlPoint, MissionTarget
 
+from ..observer import Observable
+
 if TYPE_CHECKING:
     from gen.flights.flight import FlightType
 
 
-class AirWing:
+class AirWing(Observable):
     def __init__(self, player: bool) -> None:
+        super().__init__()
         self.player = player
         self.squadrons: dict[AircraftType, list[Squadron]] = defaultdict(list)
 
     def add_squadron(self, squadron: Squadron) -> None:
+        new_aircraft_type = squadron.aircraft not in self.squadrons
+
         self.squadrons[squadron.aircraft].append(squadron)
 
+        if new_aircraft_type:
+            self.fire(type="add_aircraft_type", obj=squadron.aircraft)
+        self.fire(type="add_squadron", obj=squadron)
+
+    def remove_squadron(self, toRemove: Squadron) -> None:
+        if toRemove.aircraft in self.squadrons:
+            self.squadrons[toRemove.aircraft].remove(toRemove)
+            self.fire(type="remove_squadron", obj=toRemove)
+            if len(self.squadrons[toRemove.aircraft]) == 0:
+                self.remove_aircraft_type(toRemove.aircraft)
+
+    def remove_aircraft_type(self, toRemove: AircraftType) -> None:
+        self.squadrons.pop(toRemove)
+        self.fire(type="remove_aircraft_type", obj=toRemove)
```

Comment:
I really dislike that these can be called after game start, since they're probably all used somewhere else. I'm not sure anything reasonable can be done about that without reordering things so that the `AirWing` isn't created until after it's configured, which would be better, but probably a tough change? Any ideas, or are we probably just stuck with this?